### PR TITLE
Add CCResult.iter_err

### LINKED ISSUE #238: CCResult.iter_error
I think `iter_error : ('err -> unit) -> ('a, 'err) result -> unit` would be a nice addition to `CCResult`. I often find myself writing it by hand. I feel like it's a common pattern, especially helpful for error logging.

I'm happy to contribute if you feel like it would be a worthy addition!

### PR DESCRIPTION
**Pull request by NathanReb**
Closes #238

There you go!

The test is a bit verbose and might be a little overkill. Let me know if you'd like me to try and make it more concise, I'm not super familiar with `qtest` yet!

### CODE REVIEW

File: src/core/CCResult.mli [hunk 1]
```diff
@@ -73,6 +73,10 @@ val map2 : ('a -> 'b) -> ('err1 -> 'err2) -> ('a, 'err1) t -> ('b, 'err2) t
 val iter : ('a -> unit) -> ('a, _) t -> unit
 (** Apply the function only in case of [Ok]. *)
 
+val iter_err : ('err -> unit) -> (_, 'err) t -> unit
+(** Apply the function in case of [Error].
+    @since 2.4 *)
```
Comment (c-cube) [inline src/core/CCResult.mli]:
```suggestion
    @since NEXT_RELEASE *)
```

**Comment by c-cube** [comment]
That looks good! Just a wee change related to `@since` (and add yourself to the authors file if not there already) :)

**Comment by NathanReb** [comment]
Just did!

**Comment by c-cube** [comment]
perfect, thank you! :ok_hand:

**Pull request merged**